shared/communication/base-communication.ts

```diff
@@ -51,9 +55,13 @@ export abstract class BaseCommunication {
       this.onResponse(message, request);
     } else {
       const actOn = this.subscribers.get(message.operation);
-      const payload = await actOn?.(message.payload);
-      if (payload) {
-        await this.send({ ...message, payload });
+      try {
+        const payload = await actOn?.(message.payload);
+        if (payload) {
+          await this.send({ ...message, payload });
+        }
+      } catch (e) {
+        this.logger.error(`Exception: ${e}`);
       }
```

Comment:
Why not put the try-catch all around?

Reply:
Because I thought only this has the possibility of acting bad or going wrong